include/aikido/control/JointCommandExecutor.hpp

```diff
@@ -0,0 +1,55 @@
+#ifndef AIKIDO_CONTROL_JOINTCOMMANDEXECUTOR_HPP_
+#define AIKIDO_CONTROL_JOINTCOMMANDEXECUTOR_HPP_
+
+#include <chrono>
+#include <future>
+#include <set>
+
+#include "aikido/common/pointers.hpp"
+#include "aikido/control/Executor.hpp"
+
+namespace aikido {
+namespace control {
+
+/// Abstract class for executing joint velocity commands
+template <ExecutorType T>
+class JointCommandExecutor : public Executor
+{
+public:
+  /// Velocity-specific constructor.
+  /// \param[in] joints Vector of joint names this Executor acts upon
+  JointCommandExecutor(std::vector<std::string> joints) : Executor(T, joints)
+  {
+  }
+
+  virtual ~JointCommandExecutor()
+  {
+  }
+
+  /// Execute a Joint Command, setting future upon completion
```

Comment:
Implementation-dependent. In general, completion could mean error or success. A velocity command could return an error (e.g. reach joint limits, other controller fault), but by default, it will run until `timeout` completes.

Reply:
Timeout is technically correct in all the implementations so far.

For a position executor, it is a timeout in the traditional sense (i.e. if we don't get to the position in time, we abort).

For velocity/effort, it does function more like a duration. *However*, it is a maximum duration, since the command could be aborted or overridden by future execute commands. In the case of, say, visual servoing, it will actually function more like a timeout (as we will be sending updated commands regularly).

I think timeout makes the most sense given this.